config/katello.migrations/171030135054-foreman-proxy-update.rb

```diff
@@ -0,0 +1,6 @@
+# foreman_proxy defaults
+answers['foreman_proxy'] = {
+  'http' => true,
+  'http_port' => '8000',
+  'templates' => true
+}
```

Comment:
This will overwrite everything in `answers['foreman_proxy']` -- note how you are saying take this key in the `answers` hash and set it equal to the hash on the right side. Instead you want to update the individual values within the hash:

`answers['foreman_proxy']['http'] = true if answers['foreman_proxy'].key?('http')`